src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -187,10 +188,12 @@ def _upgrade_on_windows():
     """
     logger.warning("Updating Azure CLI with MSI from https://aka.ms/installazurecliwindows")
     tmp_dir, msi_path = _download_from_url('https://aka.ms/installazurecliwindows')
-
     logger.warning("Installing MSI")
     import subprocess
-    exit_code = subprocess.call(['msiexec.exe', '/i', msi_path])
+    subprocess.Popen(['msiexec.exe', '/i', msi_path])
+    logger.warning("Installation started, please wait for a few minutes.")
+    import sys
+    sys.exit(0)
 
     if exit_code:
```

Comment:
These below lines can never be reached, right?

Reply:
Yes, they are not reachable. This is draft PR for test only currently.